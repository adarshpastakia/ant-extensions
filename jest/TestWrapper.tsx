// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import i18next from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18next
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    resources: {
      en: {}
    },
    defaultNS: "core",
    fallbackLng: ["en"],
    keySeparator: ".",

    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  });
i18next.languages = ["en", "ar"];

export const TestWrapper: React.FC = ({ children }) => {
  return <I18nextProvider i18n={i18next}>{children}</I18nextProvider>;
};
