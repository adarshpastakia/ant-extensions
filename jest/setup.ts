// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import "@testing-library/jest-dom";
import "jsdom-global";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
});
