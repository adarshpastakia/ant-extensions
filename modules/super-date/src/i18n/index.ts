// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import i18next from "i18next";
import en from "./en.json";
import ar from "./ar.json";
import { I18nKey } from "../utils/i18nKey";

const callback = () => {
  i18next.addResourceBundle("en", I18nKey, en, true, false);
  i18next.addResourceBundle("ar", I18nKey, ar, true, false);
};

if (i18next.isInitialized) {
  callback();
} else {
  i18next.on("initialized", callback);
}
