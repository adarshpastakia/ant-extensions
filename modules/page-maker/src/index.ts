// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import "./i18n";

export { EnumTypes } from "./utils/types";
export type { PageConfig, PageItem, WidgetObject } from "./utils/types";
export { PageMaker } from "./components/PageMaker";
