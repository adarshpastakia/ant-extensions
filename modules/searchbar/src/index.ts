// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export {Type, Operator} from "./utils/types";
export type { IFilterObject, IFilterField } from "./utils/types";
export { SearchBar } from "./components/SearchBar";
