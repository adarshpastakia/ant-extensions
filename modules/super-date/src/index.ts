// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export { InlineDatePicker } from "./components/InlineDatePicker";
export { InlineRangePicker } from "./components/InlineRangePicker";
export { RelativeDatePicker } from "./components/RelativeDatePicker";
export { RelativeRangePicker } from "./components/RelativeRangePicker";
export { DateUtils } from "./utils/dateUtils";
export { DateParts } from "./utils/types";
