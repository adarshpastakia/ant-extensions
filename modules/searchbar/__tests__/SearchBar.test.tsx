// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import { render, RenderResult } from "@testing-library/react";
import React from "react";
import { TestWrapper } from "../../../jest/TestWrapper";
import { EnumFieldType, EnumOperator, SearchBar } from "../src";

describe("RelativeDatePicker", () => {
  let fragment: RenderResult;
  const fields = [
    {
      key: "string",
      type: EnumFieldType.STRING,
      defaultOperator: EnumOperator.IN,
      name: "String"
    }
  ];

  beforeEach(() => {
    fragment = render(<SearchBar collapsed={false} filters={[]} fields={fields} />, {
      wrapper: TestWrapper
    });
  });

  afterAll(() => {
    fragment.unmount();
  });

  it("should render", (done) => {
    expect(fragment.container).toMatchSnapshot();
    done();
  });

  it.todo("should fire change event");
});
