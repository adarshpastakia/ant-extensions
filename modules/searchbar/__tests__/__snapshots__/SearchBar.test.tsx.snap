// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<div>
  <div
    class="ant-ext-sb__wrapper"
  >
    <div
      class="ant-ext-sb__searchBar"
    >
      <span
        class="ant-input-group ant-input-group-compact ant-ext-sb__inputGroup"
      >
        <button
          class="ant-btn ant-ext-sb__filterToggle ant-btn-primary ant-btn-background-ghost"
          type="button"
        >
          <span
            class="ant-badge ant-badge-not-a-wrapper"
          >
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="4"
            >
              0
            </sup>
          </span>
          <span
            class="ant-ext-sb__filterToggle--label"
          >
            Filters
            Â 
          </span>
        </button>
        <span
          class="ant-input-group ant-input-group-compact ant-ext-sb__searchInput"
        >
          <span
            class="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
          >
            <span
              class="ant-input-wrapper ant-input-group"
            >
              <input
                class="ant-input"
                placeholder="Search term..."
                type="search"
                value=""
              />
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn ant-input-search-button ant-btn-primary"
                  type="button"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      class="ant-ext-sb__filterBar"
    >
      <button
        class="ant-btn ant-dropdown-trigger ant-btn-link ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="setting"
          class="anticon anticon-setting"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="setting"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
            />
          </svg>
        </span>
      </button>
      <span
        class="ant-tag ant-tag-blue ant-ext-sd__tag"
        data-testid="input-el"
      >
        <span
          aria-label="field-time"
          class="anticon anticon-field-time"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="field-time"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M945 412H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h256c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM811 548H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h122c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM477.3 322.5H434c-6.2 0-11.2 5-11.2 11.2v248c0 3.6 1.7 6.9 4.6 9l148.9 108.6c5 3.6 12 2.6 15.6-2.4l25.7-35.1v-.1c3.6-5 2.5-12-2.5-15.6l-126.7-91.6V333.7c.1-6.2-5-11.2-11.1-11.2z"
            />
            <path
              d="M804.8 673.9H747c-5.6 0-10.9 2.9-13.9 7.7a321 321 0 01-44.5 55.7 317.17 317.17 0 01-101.3 68.3c-39.3 16.6-81 25-124 25-43.1 0-84.8-8.4-124-25-37.9-16-72-39-101.3-68.3s-52.3-63.4-68.3-101.3c-16.6-39.2-25-80.9-25-124 0-43.1 8.4-84.7 25-124 16-37.9 39-72 68.3-101.3 29.3-29.3 63.4-52.3 101.3-68.3 39.2-16.6 81-25 124-25 43.1 0 84.8 8.4 124 25 37.9 16 72 39 101.3 68.3a321 321 0 0144.5 55.7c3 4.8 8.3 7.7 13.9 7.7h57.8c6.9 0 11.3-7.2 8.2-13.3-65.2-129.7-197.4-214-345-215.7-216.1-2.7-395.6 174.2-396 390.1C71.6 727.5 246.9 903 463.2 903c149.5 0 283.9-84.6 349.8-215.8a9.18 9.18 0 00-8.2-13.3z"
            />
          </svg>
        </span>
        <span>
          This week
        </span>
      </span>
      <span
        class="ant-tag ant-tag-red ant-ext-sb__filterTag"
        data-active="true"
        data-negative="true"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          style="grid-area: check;"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-ext-sb__filterTag--label ant-dropdown-trigger"
        >
          My Label87
        </span>
        <span
          aria-label="close"
          class="anticon anticon-close ant-tag-close-icon"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </span>
      </span>
      <span
        class="ant-tag ant-tag-blue ant-ext-sb__filterTag"
        data-active="true"
        data-negative="false"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          style="grid-area: check;"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-ext-sb__filterTag--label ant-dropdown-trigger"
        >
          My Label
        </span>
        <span
          aria-label="close"
          class="anticon anticon-close ant-tag-close-icon"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </span>
      </span>
      <span
        class="ant-tag ant-tag-geekblue ant-ext-sb__filterTag"
        data-active="true"
        data-negative="false"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          style="grid-area: check;"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-ext-sb__filterTag--label ant-dropdown-trigger"
        >
          <span
            class="ant-ext-sb__filterTag--clip"
          >
            String
          </span>
          <b>
            Â 
            Includes
            Â 
          </b>
          <span
            class="ant-ext-sb__filterTag--clip"
          >
            string2
          </span>
        </span>
        <span
          aria-label="close"
          class="anticon anticon-close ant-tag-close-icon"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </span>
      </span>
      <span
        class="ant-tag ant-ext-sb__addFilter ant-dropdown-trigger"
      >
        Add Filter
      </span>
    </div>
  </div>
</div>
`;
